src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionStringBuilder.cs

```diff
@@ -912,17 +911,19 @@ public override StandardValuesCollection GetStandardValues(ITypeDescriptorContex
             }
         }
 #else    
-        private static readonly string[] s_notSupportedKeywords = new string[DeprecatedKeywordsCount] {
+        private static readonly string[] s_notSupportedKeywords = {
```

Comment:
Array initialization is always the same size as the number of elements, so its unnecessary to specify it.

Reply:
Since you are eliminating the DeprecatedKeywordsCount reference here, you should just get rid of DeprecatedKeywordsCount entirely and just reference the size of this array in the one other place it's used. Otherwise, DeprecatedKeywordsCount needs to be changed from 3 to 5.

Reply:
@David-Engel Ok, I've updated the PR, chaning `DeprecatedKeywordsCount` to 5 and removing `DeprecatedSynonymsCount` from `SqlConnectionString`. As far as I can tell, I can't access `s_notSupportedKeywords`  array from `SqlConnectionString`, so I think it makes more sense to just change the number.

As far as I can tell, the only reason for calculating the count is to determine the optimum size of the dictionary. net8 added support for a `FrozenDictionary` that has its size fixed at initialization, but it looks like we can't utilize that today. That'd be a nice option here...

Please let me know if this isn't what you had in mind.

Reply:
I wasn't considering DeprecatedSynonymCount, just DeprecatedKeywordsCount. Just make sure the count is correct in the debug assert at line 927/928 of SqlConnectionString.cs after your changes. I'm pretty sure it's there to ensure we are initializing the Dictionary to the correct size to maximize perf. An incorrect count won't show up at runtime or in tests.
`Debug.Assert(synonyms.Count == count, "incorrect initial ParseSynonyms size");`

Reply:
Right. It sounds like we're on the same page, then.